stl/inc/xstring

```diff
@@ -2432,6 +2438,33 @@ public:
         _Proxy._Release();
     }
 
+    explicit basic_string(_String_constructor_concat_tag, const basic_string& _Alsource, const _Elem* const _Left_ptr,
+        const size_type _Left_size, const _Elem* const _Right_ptr, const size_type _Right_size)
+        : _Mypair(
+            _One_then_variadic_args_t(), _Alty_traits::select_on_container_copy_construction(_Alsource._Getal())) {
+        _STL_INTERNAL_CHECK(_Left_size < max_size());

```

Comment:
I believe our usual convention is to permit sizes equal to `max_size()` as suggested by the name. It should be acceptable to concatenate maximum and zero. Note that your third check permits equality.

Reply:
Hmm interesting. This is a bug in the `_STL_INTERNAL_CHECK` but not in the product and suggests I need to add zero-length appends to the test I added.